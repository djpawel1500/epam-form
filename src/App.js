import React from 'react'
import NewEventForm from './features/newEventForm/NewEventForm'
import './App.scss'

function App() {
	return (
		<>
			<NewEventForm />
		</>
	)
}

export default App
