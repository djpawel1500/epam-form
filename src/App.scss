@import 'variables.scss';

* {
	box-sizing: border-box;
	font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto',
		'Oxygen', 'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans',
		'Helvetica Neue', sans-serif;
	-webkit-font-smoothing: antialiased;
	-moz-osx-font-smoothing: grayscale;
}

body {
	background-color: $background-color;
}
input[type='text'],
textarea,
button {
	appearance: none;
}
select {
	background-color: white;
}
