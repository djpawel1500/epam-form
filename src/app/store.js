import { configureStore } from '@reduxjs/toolkit'
import titleReducer from '../features/newEventForm/title/titleSlice'
import descriptionReducer from '../features/newEventForm/description/descriptionSlice'
import categoryReducer from '../features/newEventForm/category/categorySlice'
import paymentReducer from '../features/newEventForm/payment/paymentSlice'
import rewardReducer from '../features/newEventForm/reward/rewardSlice'
import coordinatorReducer from '../features/newEventForm/coordinator/coordinatorSlice'
import durationReducer from '../features/newEventForm/duration/durationSlice'
import startsonReducer from '../features/newEventForm/startson/startsonSlice'

export default configureStore({
	reducer: {
		title: titleReducer,
		description: descriptionReducer,
		category: categoryReducer,
		payment: paymentReducer,
		reward: rewardReducer,
		coordinator: coordinatorReducer,
		duration: durationReducer,
		startson: startsonReducer,
	},
})
