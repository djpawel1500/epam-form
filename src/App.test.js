import React from 'react'
import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import store from './app/store'
import App from './App'

test('renders form submit', () => {
	const tree = render(
		<Provider store={store}>
			<App />
		</Provider>
	)
	expect(tree).toMatchSnapshot()
	// console.log(tree.toJSON())
	// const { getByText } = render(
	// 	<Provider store={store}>
	// 		<App />
	// 	</Provider>
	// )

	// expect(getByText(/Publish event/i)).toBeInTheDocument()
})
