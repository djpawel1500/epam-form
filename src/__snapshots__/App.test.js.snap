// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders form submit 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      >
        <h1>
          New event
        </h1>
      </header>
      <form
        autocomplete="off"
        class="form"
      >
        <div
          class="section"
        >
          <h2>
            About
          </h2>
          <div
            class="group"
          >
            <label
              class="label error"
              for="title"
            >
              Title*
            </label>
            <input
              class="error"
              id="title"
              name="title"
              placeholder="Make it short and clear"
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="group"
          >
            <label
              class="label error"
              for="description"
            >
              Description*
            </label>
            <div
              class="formControl"
            >
              <textarea
                class="error"
                id="description"
                maxlength="140"
                name="description"
                placeholder="Write about your event, be creative"
                required=""
              />
              <small>
                0/140
              </small>
            </div>
          </div>
          <div
            class="group"
          >
            <label
              class="label error"
              for="category"
            >
              Category
            </label>
            <div
              class="loading"
            >
              Loading...
            </div>
            <select
              class="notLoaded"
              id="category"
              name="category"
              placeholder="Select category"
              required=""
            >
              <option
                value=""
              />
            </select>
          </div>
          <div
            class="group"
          >
            <div
              class="label"
            >
              Payment
            </div>
            <div
              class="formControl"
            >
              <label
                class="radio"
              >
                <input
                  checked=""
                  name="paid"
                  type="radio"
                  value="false"
                />
                Free event
              </label>
              <label
                class="radio"
              >
                <input
                  name="paid"
                  type="radio"
                  value="true"
                />
                Paid event
              </label>
            </div>
          </div>
          <div
            class="group"
          >
            <div
              class="label"
            >
              Reward
            </div>
            <div
              class="formControl"
            >
              <input
                id="reward"
                min="0"
                name="reward"
                placeholder="Number"
                step="1"
                type="number"
                value=""
              />
              <label
                class="reward"
              >
                reward points for attendance
              </label>
            </div>
          </div>
        </div>
        <div
          class="section"
        >
          <h2>
            Coordinator
          </h2>
          <div
            class="group"
          >
            <label
              class="label error"
              for="coordinator"
            >
              Responsible*
            </label>
            <div
              class="loading"
            >
              Loading...
            </div>
            <select
              class="notSelected error notLoaded"
              id="coordinator"
              name="cordinator"
              required=""
            >
              <option
                class="notSelected"
                value=""
              >
                Select cordinator
              </option>
            </select>
          </div>
          <div
            class="group"
          >
            <label
              class="label"
              for="email"
            >
              Email
            </label>
            <input
              id="email"
              name="email"
              placeholder="Email"
              readonly=""
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="section"
        >
          <h2>
            When
          </h2>
          <div
            class="group"
          >
            <label
              class="label error"
              for="date"
            >
              Starts on*
            </label>
            <div
              class="formControl"
            >
              <input
                class="error"
                id="date"
                name="date"
                required=""
                type="date"
                value=""
              />
              <label>
                at
                <input
                  class="error"
                  id="time"
                  name="time"
                  required=""
                  type="time"
                  value=""
                />
              </label>
            </div>
          </div>
          <div
            class="group"
          >
            <label
              class="label"
              for="duration"
            >
              Duration
            </label>
            <div
              class="formControl"
            >
              <input
                id="duration"
                min="0"
                name="duration"
                placeholder="Number"
                step="0.5"
                type="number"
                value=""
              />
              <label
                class="duration"
              >
                hours
              </label>
            </div>
          </div>
        </div>
        <div
          class="submit"
        >
          <button
            type="submit"
          >
            Publish event
          </button>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    >
      <h1>
        New event
      </h1>
    </header>
    <form
      autocomplete="off"
      class="form"
    >
      <div
        class="section"
      >
        <h2>
          About
        </h2>
        <div
          class="group"
        >
          <label
            class="label error"
            for="title"
          >
            Title*
          </label>
          <input
            class="error"
            id="title"
            name="title"
            placeholder="Make it short and clear"
            required=""
            type="text"
            value=""
          />
        </div>
        <div
          class="group"
        >
          <label
            class="label error"
            for="description"
          >
            Description*
          </label>
          <div
            class="formControl"
          >
            <textarea
              class="error"
              id="description"
              maxlength="140"
              name="description"
              placeholder="Write about your event, be creative"
              required=""
            />
            <small>
              0/140
            </small>
          </div>
        </div>
        <div
          class="group"
        >
          <label
            class="label error"
            for="category"
          >
            Category
          </label>
          <div
            class="loading"
          >
            Loading...
          </div>
          <select
            class="notLoaded"
            id="category"
            name="category"
            placeholder="Select category"
            required=""
          >
            <option
              value=""
            />
          </select>
        </div>
        <div
          class="group"
        >
          <div
            class="label"
          >
            Payment
          </div>
          <div
            class="formControl"
          >
            <label
              class="radio"
            >
              <input
                checked=""
                name="paid"
                type="radio"
                value="false"
              />
              Free event
            </label>
            <label
              class="radio"
            >
              <input
                name="paid"
                type="radio"
                value="true"
              />
              Paid event
            </label>
          </div>
        </div>
        <div
          class="group"
        >
          <div
            class="label"
          >
            Reward
          </div>
          <div
            class="formControl"
          >
            <input
              id="reward"
              min="0"
              name="reward"
              placeholder="Number"
              step="1"
              type="number"
              value=""
            />
            <label
              class="reward"
            >
              reward points for attendance
            </label>
          </div>
        </div>
      </div>
      <div
        class="section"
      >
        <h2>
          Coordinator
        </h2>
        <div
          class="group"
        >
          <label
            class="label error"
            for="coordinator"
          >
            Responsible*
          </label>
          <div
            class="loading"
          >
            Loading...
          </div>
          <select
            class="notSelected error notLoaded"
            id="coordinator"
            name="cordinator"
            required=""
          >
            <option
              class="notSelected"
              value=""
            >
              Select cordinator
            </option>
          </select>
        </div>
        <div
          class="group"
        >
          <label
            class="label"
            for="email"
          >
            Email
          </label>
          <input
            id="email"
            name="email"
            placeholder="Email"
            readonly=""
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="section"
      >
        <h2>
          When
        </h2>
        <div
          class="group"
        >
          <label
            class="label error"
            for="date"
          >
            Starts on*
          </label>
          <div
            class="formControl"
          >
            <input
              class="error"
              id="date"
              name="date"
              required=""
              type="date"
              value=""
            />
            <label>
              at
              <input
                class="error"
                id="time"
                name="time"
                required=""
                type="time"
                value=""
              />
            </label>
          </div>
        </div>
        <div
          class="group"
        >
          <label
            class="label"
            for="duration"
          >
            Duration
          </label>
          <div
            class="formControl"
          >
            <input
              id="duration"
              min="0"
              name="duration"
              placeholder="Number"
              step="0.5"
              type="number"
              value=""
            />
            <label
              class="duration"
            >
              hours
            </label>
          </div>
        </div>
      </div>
      <div
        class="submit"
      >
        <button
          type="submit"
        >
          Publish event
        </button>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
